applications/DEMApplication/python_scripts/analytic_tools/analytic_data_procedures.py

```diff
@@ -3,26 +3,25 @@
 from KratosMultiphysics import *
 from KratosMultiphysics.DEMApplication import *
 
-class ParticleWatcherAnalyzer:
-    def __init__(self, analytic_particle_watcher, path, do_clear_data = True):
-        self.particle_watcher = analytic_particle_watcher
+class SurfaceAnalyzer:
+    def __init__(self, smp):
+        self.inlet = None
+        self.n_particles_accumulated = 0
+        self.mass_accumulated = 0.0
+        self.smp_name = smp.Name
 
-    # call everytime post results are generated
-    def SetNodalMaxImpactVelocities(self, analytic_model_part):
-        self.analytic_particle_watcher.SetNodalMaxImpactVelocities(analytic_model_part)
+        self.ghost = smp[IS_GHOST]
```

Comment:
not necessary, this SurfaceAnalyzer is always ghost

Reply:
Flag was legacy but I will verify it is always true.
 @jgonzalezusua

Reply:
@farrufat-cimne you are only creating this object when IS_GHOST:
```
for smp in sub_model_parts:
            if smp[IS_GHOST] == True:
                self.ghost_smp_detected = True
                self.surface_analyzers_list.append(SurfaceAnalyzer(smp))
```